src/UsgsAstroLsSensorModel.cpp

```diff
@@ -870,22 +873,27 @@ double UsgsAstroLsSensorModel::getImageTime(
 {
    // Flip image taken backwards
    double line1 = image_pt.line;
-   if (_data.m_ImageFlipFlag == 1)
-      line1 = _data.m_TotalLines - 1 - image_pt.line;
 
    // CSM image convention: UL pixel center == (0.5, 0.5)
    // USGS image convention: UL pixel center == (1.0, 1.0)
 
    double lineCSMFull = line1 + _data.m_OffsetLines;
    double lineUSGSFull = lineCSMFull + 0.5;
 
-   // USGS timing reference leftmost-side 1st full row
-   double timeUsgs = _data.m_IntTime * (lineUSGSFull - 0.5)
-      + _data.m_StartingEphemerisTime;
+   // These calculation assumes that the values in the integration time
+   // vectors are in terms of ISIS' pixels
+   auto referenceLineIt = std::upper_bound(_data.m_IntTimeLines.begin(),
+                                           _data.m_IntTimeLines.end(),
+                                           lineUSGSFull);
+   if (referenceLineIt != _data.m_IntTimeLines.begin()) {
+      --referenceLineIt;
+   }
+   size_t referenceIndex = std::distance(_data.m_IntTimeLines.begin(), referenceLineIt);

```

Comment:
If the exposure time table was a vector of structs/arrays instead of 3 vectors, we wouldn't need to convert the iterator to an index, we could just use the iterator directly for the time calculation.

Reply:
:+1: go with a vector of structs.